nexus/reconfigurator/execution/src/omicron_zones.rs

```diff
@@ -178,6 +181,42 @@ pub(crate) async fn clean_up_expunged_zones<R: CleanupResolver>(
     }
 }
 
+async fn oximeter_cleanup(
+    opctx: &OpContext,
+    datastore: &DataStore,
+    zone_id: OmicronZoneUuid,
+    log: &Logger,
+) -> anyhow::Result<()> {
+    // Record that this Oximeter instance is gone.
```

Comment:
FYI @sunshowers this has some of the same issues with cleanup that we've been talking about with external networking, although I _think_ in this case there's an easier path forward because we're soft deleting. "Is cleanup from oximeter expungement done" should check:

* Does the `oximeter` row have a non-NULL `time_expunged`? (If "yes", proceed to the next check.)
* Are there any rows in `metric_producer` that have an `oximeter_id` matching our expunged zone? (If "no", we can clean up.)

It is possible for the answer to the second question to be "yes" for an arbitrary amount of time after the first question is "yes", because although we mark the Oximeter expunged immediately, we can't successfully reassign any producers until another Oximeter is up and running. Once the first question is "yes", though, it's not possible for _new_ entries to show up in `metric_producer` with that Oximeter ID (checked in the insert query [here](https://github.com/oxidecomputer/omicron/blob/db6ea7effa67c3a1a1f82793c970c5f75694cf56/nexus/db-queries/src/db/datastore/oximeter.rs#L212-L216)).